core/src/main/java/io/github/tcdl/msb/api/MessageTemplate.java

```diff
@@ -28,8 +47,32 @@ public MessageTemplate withTtl(Integer ttl) {
         return this;
     }
 
+    /**
+     * @return tags of the message
+     */
+    public List<String> getTags() {
+        return tags;
+    }
+
+    /**
+     * @param tags of the message
+     */
+    public MessageTemplate withTag(String... tags) {
+        this.tags = Arrays.asList(tags);
+        return  this;
+    }
+
+    public MessageTemplate addTag(String... tags) {
+        this.tags.addAll(Arrays.asList(tags));
+        return  this;
+    }
+
+    public static MessageTemplate copyOf(MessageTemplate from) {
```

Comment:
Why do we need this method? Why can't we use the constructor directly?

If we do need it then let's keep it near constructors.